### DISCUSSION TITLE
Plugin-Dev Feedback

### DISCUSSION THREAD

**Discussion by LordTylus** [apeira/Torch#17] [General]
Hey there,

as requested i got some feedback to the Plugin dev section as is. 

I like the folder structure starting with the yaml and having a dedicated space for assets is nice too. Question now is, will the plugin loader be more smart then in Torch 1? Like how does it behave on conflicts, when two different versions of the same DLL are present in two plugins? 

Additionally I am missing a feature in the plugins yet. I do not want hard requirements but also have soft dependencies which are just loaded first if available but dont have to be there if they arent. 

Next I havent quite understood the example with Command and Permissions yet. It looks a bit complicated. As I pretty much want to continue using the permissions SE provides. While torch 2 is a bit more powerful and not bound to SE I would still like to know how I can pull that off if I want.

**Comment by Jimmacle** [comment]
The new loader won't blindly load every assembly in a plugin package, only the entry point DLL specified in the YAML file. The rest will be left up to the assembly resolver which will hopefully Just Work:tm:. I've updated the wiki and prototype installer code to support defining incompatible plugins and specifying a soft load order.

The command and permissions systems are implemented in a very generic way so they can be adapted to any game. I'm planning for the SE core plugin to do some finessing of the system to create predefined permission groups matching the ingame ranks. The command system is also more generic and doesn't currently have features for automatic argument parsing, though that is something that can be added.
